tests/ignite/metrics/test_root_mean_squared_error.py

```diff
@@ -40,23 +50,10 @@ def _test(y_pred, y, batch_size):
         assert isinstance(res, float)
         assert pytest.approx(res) == np_res
 
-    def get_test_cases():
-
-        test_cases = [
-            (torch.empty(10).uniform_(0, 10), torch.empty(10).uniform_(0, 10), 1),
-            (torch.empty(10, 1).uniform_(-10, 10), torch.empty(10, 1).uniform_(-10, 10), 1),
-            # updated batches
-            (torch.empty(50).uniform_(0, 10), torch.empty(50).uniform_(0, 10), 16),
-            (torch.empty(50, 1).uniform_(-10, 10), torch.empty(50, 1).uniform_(-10, 10), 16),
-        ]
-
-        return test_cases
-
     for _ in range(5):
         # check multiple random inputs as random exact occurencies are rare
-        test_cases = get_test_cases()
-        for y_pred, y, batch_size in test_cases:
-            _test(y_pred, y, batch_size)
+        (y_pred, y, batch_size) = test_cases
```

Comment:
@nmcguire101 If you take a look attentively what does `test_cases = get_test_cases()` previously, you can see that in the loop `for _ in range(5):` all `test_cases` are different. In your PR however, they are all the same and thus there is no point to make the loop.
Can you check however whether `test_cases` become different if you add another dummy arg to `@pytest.mark.parametrize` to represent the loop `for _ in range(5):` ?

Reply:
@nmcguire101 Can you check however whether `test_cases` are different with `dummy_val` ? Thanks

Reply:
I realized there was an error in my last PR, this latest one adds a few lines but generates unique values. This fix may be trivial as it only gets rid of two lines. If that's the case I'll keep looking for other tests to improve.

Reply:
I see, I also checked that parametrize does not generate different data for different with `dummy_val`.
As commented previously, the idea of using pytest.mark.parametrize to get rid of _test inner method.
Can you check fixture parametrize: https://docs.pytest.org/en/6.2.x/fixture.html#fixture-parametrize to see if we could get what we want: 1) get rid of `_test`, 2) have different test values (`y_pred, y, batch_size`) and 3) run the check with a parametrized loop for 3 times ?

Reply:
I was able to get rid of _test and generate unique test values for `y_pred,` y, batch_size`. However, I am confused what you mean by run the check with a parametrized loop for 3 times because based on the original code it should be run 5 times (unless I am misunderstanding what you're referring to here). I will submit a new PR after getting clarification on the third point.

Reply:
I was able to get rid of _test and generate unique test values for `y_pred,` `y,` `batch_size`. However, I am confused what you mean by run the check with a parametrized loop for 3 times because based on the original code it should be run 5 times (unless I am misunderstanding what you're referring to here). I will submit a new PR after getting clarification on the third point.

Reply:
Sorry, i wanted just to reduce the number of times to run the test. 5 -> 3

Reply:
@nmcguire101 do you think we can get rid of `for test in generate_tests:` loop using fixture parametrize ?

Reply:
I'm not sure that would work without defining four different fixtures, one for each test case. Do you have a different idea?

Reply:
I took another look at it and found a way to get rid of the for loop.